vgrep.go

```diff
@@ -89,13 +92,16 @@ func main() {
 	// interactive sheel.
 	if Options.Show != "" || Options.Interactive {
 		commandParse(Options.Show)
+		waiter.Wait()
```

Comment:
That's not (yet) completely protecting against the bug as the waiting group does not sync across processes. Instead, I suggest using a file lock (https://github.com/nightlyone/lockfile) to sync read/write on the cache. The library isn't perfect as a PID isn't unique when a previous owner died but it's good enough for a single user :) A good place for the lockfile could be `./local/share/vgrep/cache-lock`.

In theory, the filelock could still cause ordering issues when multiple processes read from and write to the cache. But I hope that a single user won'tl try DOSing vgrep, and in case we can add a `--no-cache` option.